File: q1287/q1287.py
class Solution(object):
    def findSpecialInteger(self, arr):
        """
        :type arr: List[int]
        :rtype: int
        """
        i = 0
        n = len(arr) // 4
        while i < len(arr) - n: 
            if arr[i] == arr[i+n]: return arr[i]
            i += 1
        

File: q1287/q1287.cpp
class Solution
{
public:
    int findSpecialInteger(vector<int> &arr)
    {
        int i = 0;
        int n = arr.size() / 4;
        while (i < arr.size() - n)
        {
            if (arr[i] == arr[i + n])
                return arr[i];
            i++;
        }
        return -1;
    }
};